homeassistant/components/kraken/sensor.py

```diff
@@ -0,0 +1,230 @@
+            return self.coordinator.data[self.tracked_asset_pair_wsname]["bid"][1]
+        if self._sensor_type == "volume_today":
+            return self.coordinator.data[self.tracked_asset_pair_wsname]["volume"][0]
+        if self._sensor_type == "volume_last_24h":
+            return self.coordinator.data[self.tracked_asset_pair_wsname]["volume"][1]
+        if self._sensor_type == "volume_weighted_average_today":
+            return self.coordinator.data[self.tracked_asset_pair_wsname][
+                "volume_weighted_average"
+            ][0]
+        if self._sensor_type == "volume_weighted_average_last_24h":
+            return self.coordinator.data[self.tracked_asset_pair_wsname][
+                "volume_weighted_average"
+            ][1]
+        if self._sensor_type == "number_of_trades_today":
+            return self.coordinator.data[self.tracked_asset_pair_wsname][
+                "number_of_trades"
+            ][0]
+        if self._sensor_type == "number_of_trades_last_24h":
+            return self.coordinator.data[self.tracked_asset_pair_wsname][
+                "number_of_trades"
+            ][1]
+        if self._sensor_type == "low_today":
+            return self.coordinator.data[self.tracked_asset_pair_wsname]["low"][0]
+        if self._sensor_type == "low_last_24h":
+            return self.coordinator.data[self.tracked_asset_pair_wsname]["low"][1]
+        if self._sensor_type == "high_today":
+            return self.coordinator.data[self.tracked_asset_pair_wsname]["high"][0]
+        if self._sensor_type == "high_last_24h":
+            return self.coordinator.data[self.tracked_asset_pair_wsname]["high"][1]
+        if self._sensor_type == "opening_price_today":
+            return self.coordinator.data[self.tracked_asset_pair_wsname][
+                "opening_price"
+            ]
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        if self._target_asset == "EUR":
+            return "mdi:currency-eur"
+        if self._target_asset == "GBP":
+            return "mdi:currency-gbp"
+        if self._target_asset == "USD":
+            return "mdi:currency-usd"
+        if self._target_asset == "JPY":
+            return "mdi:currency-jpy"
+        if self._target_asset == "XBT":
+            return "mdi:currency-btc"
+        return "mdi:cash"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        if "number_of" not in self._sensor_type:
+            return self._unit_of_measurement
+
+    @property
+    def available(self):
+        """Could the api be accessed during the last update call."""
+        return self._available and self.coordinator.last_update_success
+
+    @property
+    def device_info(self) -> dict:
+        """Return a device description for device registry."""
+
+        return {
+            "identifiers": {(DOMAIN, self._source_asset, self._target_asset)},
```

Comment:
There should only be two items in the tuple. Make an f-string of the source and target assets.